### PR TITLE
Support configuring gRPC options in the Gradle plugin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 1]
```diff
@@ -60,186 +60,192 @@ sealed class Target {
     logger: WireLogger
   ): SchemaHandler
 
-  /** Generate `.java` sources. */
-  data class JavaTarget(
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: I promoted this to be top-level

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 2]
```diff
@@ -60,186 +60,192 @@ sealed class Target {
...
-      }
-    }
+  interface SchemaHandler {
+    fun handle(type: Type)
+    fun handle(service: Service)
   }
+}
 
-  /** Generate `.kt` sources. */
-  data class KotlinTarget(
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: I promoted this to be top-level

File: wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt [hunk 3]
```diff
@@ -60,186 +60,192 @@ sealed class Target {
...
+          throw IOException("Error emitting " +
+              "${kotlinFile.packageName}.${service.type()} to $outDirectory", e)
         }
       }
     }
   }
+}
 
-  /** Omit code generation for these sources. Use this for a dry-run. */
-  data class NullTarget(
```
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt]: I promoted this to be top-level

File: docs/wire_compiler.md [hunk 4]
```diff
@@ -318,8 +319,13 @@ wire {
     // target.
     javaInterop = true
 
-    // True for emitted services to provide blocking APIs only.
-    blockingServices = true
+    // SUSPENDING to generate coroutines APIs that require a Kotlin coroutines context.
+    // BLOCKING to generate blocking APIs callable by Java and Kotlin.
+    rpcCallStyle = com.squareup.wire.kotlin.RpcCallStyle.BLOCKING
```
Comment (Egorand) [inline docs/wire_compiler.md]: With no auto-complete in Groovy I expect these to be painful to use.. Should we introduce overloads that accept Strings?
```groovy
wire {
  kotlin {
    rpcCallStyle = 'blocking'
    rpcRole = 'server'
  }
}
```

Comment (swankjesse) [reply docs/wire_compiler.md]: What’s idiomatic?

Comment (swankjesse) [reply docs/wire_compiler.md]: I’m gonna land as-is and add more later.

Comment (Egorand) [reply docs/wire_compiler.md]: Yeah, not sure what the idiomatic way is.

Comment (JakeWharton) [reply docs/wire_compiler.md]: definitely a string. you can do `values().singleOrNull { it.equals(whatever, ignoreCase=true) } ?: throw IllegalArgumentException("Unknown whatever $whatever. Valid values: ${values().contentToString()}")`

**Pull request merged**